src/frontend/OverlayManager/index.tsx

```diff
@@ -1,61 +1,120 @@
-import React, { useEffect, useState } from 'react'
+import React from 'react'
 import BrowserGameStyles from './index.module.scss'
-import BrowserExtensionManager from './ExtensionManager'
-import BrowserToastManager from './ToastManager'
+import ToastManager from './ToastManager'
 import { PROVIDERS } from 'common/types/proxy-types'
-import BrowserExtensionToastManager from './ExtensionToastManager'
 import { Runner } from 'common/types'
-
-interface RenderState {
-  showToasts: boolean
-  showExtension: boolean
-  showBrowserGame: boolean
-  showExitButton: boolean
-  showHintText: boolean
-}
+import { observer } from 'mobx-react-lite'
+import OverlayState from 'frontend/state/OverlayState'
+import WalletState from 'frontend/state/WalletState'
+import { t } from 'i18next'
+import { Button } from '@hyperplay/ui'
+import DeviceState from 'frontend/state/DeviceState'
+import ExtensionManager from 'frontend/ExtensionManager'
+import TransactionState from 'frontend/state/TransactionState'
 
 interface BrowserGameProps {
-  url: string
   appName: string
   runner: Runner
-  renderState: RenderState
 }
 
-const OverlayManager = function ({
-  url,
+const Overlay = observer(function ({ appName, runner }: BrowserGameProps) {
+  const txnToastContainerStyle = {} as React.CSSProperties
+  if (OverlayState.title === 'HyperPlay Toasts') {
+    txnToastContainerStyle.bottom = 'unset'
+    txnToastContainerStyle.right = 0
+    txnToastContainerStyle.top = 0
+  }
+
+  let exitGameButtonStyle = {
+    top: 'var(--space-md)',
+    right: 'var(--space-md)',
+    position: 'absolute',
+    zIndex: 200
+  } as React.CSSProperties
+
+  if (
+    OverlayState.renderState.showExitGameButton &&
+    !OverlayState.renderState.showExtension
+  ) {
+    exitGameButtonStyle = {
+      ...exitGameButtonStyle,
+      top: 0,
+      right: 0,
+      overflowY: 'hidden'
+    }
+  }
+
+  const shouldShowExtension =
+    WalletState.provider === PROVIDERS.METAMASK_EXTENSION &&
+    OverlayState.renderState.showExtension
+
+  return (
+    <>
+      {OverlayState.renderState.showToasts ? (
+        <div
+          className={BrowserGameStyles.txnToastContainer}
+          style={txnToastContainerStyle}
+        >
+          <ToastManager />
+        </div>
+      ) : null}
+
+      {TransactionState.isInitialToastShown ? null : (
+        <>
+          <div className={BrowserGameStyles.bgFilter}></div>
+          {OverlayState.renderState.showHintText ? (
+            <div className={`${BrowserGameStyles.closeOverlayText} title`}>
+              {t('hyperplayOverlay.closeOverlay', {
+                defaultValue:
+                  'Press {{overlayKeyMod}} + X to close the overlay',
+                overlayKeyMod: DeviceState.isMac ? 'Option' : 'Alt'
+              })}
+            </div>
+          ) : null}
+          {OverlayState.renderState.showExitGameButton ? (
```

Comment:
I agree, this is confusing. Perhaps split this into smaller components.

Reply:
@eliobricenov sure I can make that change. it is ultimately a stylistic preference

@flavioislima with elio's method of assigning it to a variable, the variable name serves much the same purpose of a different component. I'll just go with that since there is no other benefit to breaking out into a separate component

Reply:
I think multiple components will make even more noise. This is not a blocker by any chance.